app/js/app/app.js

```diff
@@ -700,9 +700,17 @@ define([
 
                             if (ns.length > 0) {
 
-                                $rootScope.notificationDoc = ns[0];
+                                // dirty hack for student/teacher questionnaires, but notifications wont be like this for long (05/17)
+                                for (var i = 0; i < ns.length; i++) {
 
-                                $rootScope.modals.notification.show();
+                                    if ($rootScope.user.role.toLowerCase() == ns[i].tags[0]) {
```

Comment:
Yes, the previous incarnation would fetch an array of notifications for a user and then only show the one at index 0, so have just reused that. It would probably be a nicer idea to actually filter notifications by teacher/student in the backend anyway, so will bear that in mind for the future.